src/main/java/net/minecraftforge/common/ForgeConfigSpec.java

```diff
@@ -123,8 +125,7 @@ private int correct(Config spec, CommentedConfig config, LinkedList<String> pare
             final Object configValue = configMap.get(key);
             final CorrectionAction action = configValue == null ? ADD : REPLACE;
 
-            if (!dryRun)
-                parentPath.addLast(key);
+            parentPath.addLast(key);
```

Comment:
whats with this change?

Reply:
When `parentPath` was null, none of the levelComments were found (`levelComments.get(parentPath)` would return null), and so the comments were thought not to be matching.